### PR TITLE
MINOR: add UPGRADE_FROM to 2.0-2.3 config docs

### PR DESCRIPTION
**Pull request by ableegoldman**
[This PR](https://github.com/apache/kafka/pull/7825) but for 2.0-2.3

Targeted against 2.3, should be cherry-picked back to 2.0 after which we can merge to kafka-site

### CODE REVIEW

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -284,6 +284,11 @@ <h4><a class="toc-backref" href="#id5">bootstrap.servers</a><a class="headerlink
             <td colspan="2">Timestamp extractor class that implements the <code class="docutils literal"><span class="pre">TimestampExtractor</span></code> interface.</td>
             <td>See <a class="reference internal" href="#streams-developer-guide-timestamp-extractor"><span class="std std-ref">Timestamp Extractor</span></a></td>
           </tr>
+          <tr class="row-odd"><td>upgrade.from</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: This should be `row-even` and all the row below need to be "switched" (an annoying task... We really need to move off HTML...)

Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html]: Oh I totally missed that...wow that's awful 😬

Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html]: What difference does it make? I notice it's also screwed up between `partition.grouper` and `processing.guarantee`, and between `replication.factor` and `retries` and `retry.backoff.ms`

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html]: Well, I guess we should fix the screw up than? It should influence the coloring if the row from my understanding.

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html]: Locking at https://kafka.apache.org/documentation/streams/developer-guide/config-streams.html#optional-configuration-parameters it does actually not influence the colering... Not sure why it's there now... \cc @JimGalasyn

Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html]: Hm. Interestingly,[ the table](https://kafka.apache.org/24/documentation/streams/developer-guide/config-streams.html#optional-configuration-parameters) seems to render correctly nevertheless...I can make a ticket for it, just trying to understand what (if anything) it actually affects

**Comment by ableegoldman** [comment]
Yikes, that's a lot of test failures on the 2.3 branch...I guess we aren't backporting fixes for flaky tests?

File: docs/streams/developer-guide/config-streams.html (hunk 1 above)
Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html]: Whoops, jinx

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html]: Maybe @londoncalling can help on the even/odd pattern?

Comment (ableegoldman) [reply docs/streams/developer-guide/config-streams.html]: FWIW I did fix it in this PR, and have a followup PR to fix it in 2.4-trunk: https://github.com/apache/kafka/pull/7837

If maintaining this pattern really is important, we should still fix the other configs in separate PR(s) so we can target the right branches

**Comment by guozhangwang** [comment]
I'm fine merging it as is once @mjsax is happy with his comments -- some of the flaky tests may not be fixed in 2.3.

**Pull request merged**